src/services/ipfs/ipfs.resolver.ts

```diff
@@ -0,0 +1,34 @@
+import { IpfsUploadFailedException } from '@common/exceptions/ipfs.exception';
+import { Inject } from '@nestjs/common';
+import { Args, Mutation, Resolver } from '@nestjs/graphql';
+import { createWriteStream } from 'fs';
+import { FileUpload, GraphQLUpload } from 'graphql-upload';
+import { IpfsService } from './ipfs.service';
+import { v4 as uuidv4 } from 'uuid';
+
+@Resolver()
+export class IpfsResolver {
+  constructor(@Inject(IpfsService) private ipfsService: IpfsService) {}
+
+  @Mutation(() => String)
+  async uploadFile(
+    @Args({ name: 'file', type: () => GraphQLUpload })
+    { createReadStream, filename }: FileUpload
+  ): Promise<string> {
+    const guid = uuidv4();
+    const filePath = `./uploads/${filename}-${guid}`;
+
+    const res = new Promise(async (resolve, reject) =>
+      createReadStream()
+        .pipe(createWriteStream(filePath))
+        .on('finish', () => resolve(true))
+        .on('error', () => reject(false))
+    );
+
+    if (await res) {
```

Comment:
remove the temporary file after with ipfs upload succeeds or on an error? Otherwise this temp storage will fill up...

Reply:
Addressed in 25eb4f5a65de0e22a10bcddbc7317ac9e2360584